esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5885,4 +5885,115 @@ int CWsDfuEx::GetIndexData(IEspContext &context, bool bSchemaOnly, const char* i
+    if (!group)
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "No group found for %s.", groupName);
+    if (groupType == grp_roxie)
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "Roxie group is not allowed: %s.", groupName);
+
+    //Create a FileDescriptor
+    Owned<IFileDescriptor> fileDesc = createFileDescriptor();
+    fileDesc->queryProperties().setProp("@job", jobID);
+    if (!isEmptyString(userID))
+        fileDesc->queryProperties().setProp("@owner", userID);
+
+    StringBuffer curDir, defaultDir;
+    if (!getConfigurationDirectory(nullptr, "data", groupType == grp_hthor ? "hthor" : "thor", groupName, defaultDir))
+        makePhysicalPartName(logicalName, 0, 0, curDir, false, DFD_OSdefault); // legacy
+    else
+        makePhysicalPartName(logicalName, 0, 0, curDir, false, SepCharBaseOs(getPathSepChar(defaultDir)), defaultDir.str());
+
+    fileDesc->setDefaultDir(curDir.str());
+
+    StringBuffer partMask;
+    unsigned parts = group->ordinality();
+    getPartMask(partMask, logicalName, parts);
+    fileDesc->setNumParts(parts);
+    fileDesc->setPartMask(partMask);
+
+    ClusterPartDiskMapSpec mspec;
+    mspec.defaultCopies = DFD_DefaultCopies;
+    fileDesc->addCluster(groupName, group, mspec);
+
+    //Publish the file
+    Owned<IDistributedFile> file = queryDistributedFileDirectory().createNew(fileDesc);
+    file->setAccessed();
+    file->setECL(recordDefinition);
+    file->queryAttributes().setPropInt64("@size", 0);                
+    file->queryAttributes().setPropBin("_rtlType", layoutBin.length(), layoutBin.toByteArray());
+    file->attach(logicalName, userDesc);
+}
+
+bool CWsDfuEx::onDFUCreateAndPublish(IEspContext &context, IEspDFUCreateAndPublishRequest &req, IEspDFUCreateAndPublishResponse &resp)
+{
+    try
+    {
+        if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Write, false))
+            throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "Failed to CreateAndPublish. Permission denied.");
+
+        const char *logicalName = req.getLogicalName();
+        const char *groupName = req.getGroupName();
+        const char *recordDefinition = req.getECLRecordDefinition();
+        if (isEmptyString(logicalName))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT,"No LogicalName defined.");
+        if (isEmptyString(groupName))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT,"No GroupName defined.");
+        if (isEmptyString(recordDefinition))
+             throw MakeStringException(ECLWATCH_INVALID_INPUT,"No ECLRecordDefinition defined.");
+
+        StringBuffer userID;
+        context.getUserID(userID);
+
+        Owned<IUserDescriptor> userDesc;
+        if(!userID.isEmpty())
+        {
+            userDesc.setown(createUserDescriptor());
+            userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
+        }
+
+        const char *jobID = req.getJobID();
```

Comment:
there will now be a new input field "jobdescription" to be used here instead of jobid.